lib/Commands/InterruptSignalAwaiter.h

```diff
@@ -0,0 +1,67 @@
+//===- InterruptSignalHandler.h ---------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_COMMAND_INTERRUPTSIGNALAWAITER_H
+#define LLBUILD_COMMAND_INTERRUPTSIGNALAWAITER_H
+
+#include <atomic>
+#include <functional>
+#include <thread>
+
+namespace llbuild {
+namespace command {
+/// A platform-independent abstraction that allows users to provide a
+/// function that is called when a SIGINT signal is received by the program.
+struct InterruptSignalAwaiter {
+ private:
+  void(*previousSignalHandler)(int);
+  static int signalWatchingPipe[2];
```

Comment:
This has to be static, because it is used in the signal handler `static void sigintHandler(int);` which has to be static:

```cpp
void InterruptSignalAwaiter::sigintHandler(int) {
  // Set the atomic interrupt flag.
  wasInterrupted = true;

  // Write to wake up the signal monitoring thread.
  char byte{};
  basic::sys::write(signalWatchingPipe[1], &byte, 1);
}
```

Do correct me if I'm wrong though